src/main/java/file/TaskSaver.java

```diff
@@ -0,0 +1,50 @@
+package file;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+
+import Tasks.Task;
+import Tasks.Todo;
+import Tasks.Deadline;
+import Tasks.Event;
+
+/**
+ * This class handles the saving of task data to a file.
+ * It writes task information in a specific format so that it can later be loaded correctly.
+ */
+public class TaskSaver {
+    /**
+     * Saves a list of tasks to a file in a specific format.
+     * Each task is written as a string representing its type, completion status, and other relevant details.
+     *
+     * @param filePath the file path where the tasks should be saved
+     * @param tasks the list of tasks to save
+     * @throws IOException if there is an error while writing to the file
+     */
+    public static void saveTasks(String filePath, ArrayList<Task> tasks) throws IOException {
+        FileWriter writer = new FileWriter(filePath);
+
+        // Loop through each task and write its data in a specific format
+        for (Task task : tasks) {
+            String taskType = "";
+            String taskData = "";
+
+            if (task instanceof Todo) {
+                taskType = "T";  // Represents a Todo task
+                taskData = taskType + " | " + (task.isDone() ? "1" : "0") + " | " + ((Todo) task).getDescription();
+            } else if (task instanceof Deadline) {
+                taskType = "D";  // Represents a Deadline task
+                taskData = taskType + " | " + (task.isDone() ? "1" : "0") + " | " + ((Deadline) task).getDescription() + " | " + ((Deadline) task).getDeadlineTime();
+            } else if (task instanceof Event) {
+                taskType = "E";  // Represents an Event task
+                taskData = taskType + " | " + (task.isDone() ? "1" : "0") + " | " + ((Event) task).getDescription() + " | " + ((Event) task).getFromTime() + " | " + ((Event) task).getToTime();
```

Comment:
While this is relatively well written, i suggest you refactor these segments to form the `taskData` variable into functions of their own to avoid showing the reader so many complex expressions at once. That might help with the readability of this segment. You also run the risk of hitting >120 characters on a line so that needs to be avoided too (line 41 has 176 characters!) .